### PR TITLE
Initial RSpec implementation

### PR DESCRIPTION
**Pull request by Shmekal**
Commit description:
- remove all non-rspec test-files(scripts)
- rename spec/basic_spec.rb to spec/register_spec.rb and move it to spec/features/
- most important: reorganize register_spec.rb and respective class
  note: not sure if split checks by register is worth doing at all as register of initial query affects nothing - register of Title and Description depends only on the particular search result and URL is always downcase

Notes:
Такой громоздкой коммит получился. Впоследующем лучше разбивать на логические части?

### CODE REVIEW

File: spec/features/register_spec.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+RSpec.feature "Verify basic query search results" do
+
+  before do
+    @page = Page.new
+    @page.visit 'https://google.com'
+  end
+
+  feature "Upcase input" do
+
+    # not sure if curly bracket syntax is acceptable
```
Comment (nonkor) [inline spec/features/register_spec.rb]: yep, acceptable

File: spec/features/register_spec.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    before { @page.enter_keyword 'Monster' }
+
+    feature "First result" do
+      scenario "Title contains inputted keyword Upcase" do
+        expect(@page.first_result_title).to include('Monster')
+      end
+
+      scenario "URL contains inputted keyword Downcase" do
+        expect(@page.first_result_url).to include('monster')
```
Comment (nonkor) [inline spec/features/register_spec.rb]: why downcased version here?

File: spec/features/register_spec.rb [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  end
+
+
+  feature "Downcase input" do
+
+    before { @page.enter_keyword 'monster' }
+
+    feature "First result" do
+      scenario "Title contains inputted keyword Upcase" do
+        expect(@page.first_result_title).to include('Monster')
```
Comment (nonkor) [inline spec/features/register_spec.rb]: upcase?

File: spec/features/register_spec.rb [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+      end
+
+      scenario "URL contains inputted keyword Downcase" do
+        expect(@page.first_result_url).to include('monster')
+      end
+
+      scenario "Description contains inputted keyword Upcase" do
+        expect(@page.first_result_description).to include('Monster')
```
Comment (nonkor) [inline spec/features/register_spec.rb]: upcase?

File: lib/page.rb [hunk 5]
```diff
@@ -40,4 +35,34 @@ def parse_all_sites
...
+  def first_result_description
+    parse_first_result(:description)
+  end
+
+  private
+
+  def parse_first_result(destination)
+    locator =
+      case destination
+        when :title
```
Comment (nonkor) [inline lib/page.rb:55]: good for now, but would be better to wrap them into page objects in future.

**Review by nonkor — CHANGES_REQUESTED** [review]
Strange PR name, especially if know there is no gemfile touching :)
But in general - good PR, taka a look on my small issues.

**Comment by nonkor** [comment]
About the commit: this seems ok for me. But yeah, makes sense to granulate them by logical parts.

File: spec/features/register_spec.rb (hunk 2 above)
Comment (Shmekal) [reply spec/features/register_spec.rb]: the thing is that url register is always downcased. knowing that we always expect it that way(downcase)

File: spec/features/register_spec.rb (hunk 3 above)
Comment (Shmekal) [reply spec/features/register_spec.rb]: @nonkor , yes. the register in description and title doesn't correlate with the initial query(keyword) register. so as long as we check the hardcoded values this check expects particular register as we know what result will be shown. the test will become falling once search engine changes the first result which will include the downcased keyword. and this is the weak point I don't know how to deal with. maybe we should refuse doing register checks for the title and description sections? but probably there's already a conventional solution for such cases.

File: spec/features/register_spec.rb (hunk 4 above)
Comment (Shmekal) [reply spec/features/register_spec.rb]: @nonkor see the :41 comment

File: lib/page.rb (hunk 5 above)
Comment (Shmekal) [reply lib/page.rb:55]: @nonkor attr_accessor's, right?

**Comment by Shmekal** [comment]
@nonkor not sure how we go on. but here's the last commit description:
- remove duplicate in Gemfile(finally)
- split spec into two by the case(register) principle

File: lib/page.rb (hunk 5 above)
Comment (nonkor) [reply lib/page.rb:55]: no, I mean [page object pattern](https://habrahabr.ru/company/wapstart/blog/138674/). I would recommend to read that doc.

File: spec/features/register_spec.rb (hunk 3 above)
Comment (nonkor) [reply spec/features/register_spec.rb]: I would explain it by a call, much quicker.

File: spec/features/basic_search_spec.rb [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+        scenario "URL contains inputted keyword Downcase" do
+          expect(subject.first_result_url).to include(keyword.downcase)
+        end
+
+        scenario "Description contains inputted keyword Upcase" do
+          expect(subject.first_result_description).to include(keyword.capitalize)
+        end
+
+      end # context downcase
+
```
Comment (nonkor) [inline spec/features/basic_search_spec.rb:31]: You like empty lines so much. I recommend to use [ruby-style-guide](https://github.com/bbatsov/ruby-style-guide).

**Review by nonkor — APPROVED** [review]
I like it. Great job!
Next step: create page objects by to page object pattern. Good implementation of this pattern for capybara is [site_prism](https://github.com/natritmeyer/site_prism)

**Pull request merged**